test/e2e/e2e_duplicatepods_test.go

```diff
@@ -0,0 +1,198 @@
+		t.Fatalf("Unable to create ns %v", testNamespace.Name)
+	}
+	defer clientSet.CoreV1().Namespaces().Delete(ctx, testNamespace.Name, metav1.DeleteOptions{})
+
+	t.Log("Creating duplicates pods")
+
+	deploymentObj := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "duplicate-pod",
+			Namespace: testNamespace.Name,
+			Labels:    map[string]string{"app": "test-duplicate", "name": "test-duplicatePods"},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": "test-duplicate", "name": "test-duplicatePods"},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": "test-duplicate", "name": "test-duplicatePods"},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{{
+						Name:            "pause",
+						ImagePullPolicy: "Always",
+						Image:           "kubernetes/pause",
+						Ports:           []v1.ContainerPort{{ContainerPort: 80}},
+					}},
+				},
+			},
+		},
+	}
+
+	tests := []struct {
+		description             string
+		replicasNum             int
+		beforeFunc              func(deployment *appsv1.Deployment)
+		expectedEvictedPodCount int
+	}{
+		{
+			description: "Evict Pod even Pods schedule to specific node",
+			replicasNum: 4,
+			beforeFunc: func(deployment *appsv1.Deployment) {
+				deployment.Spec.Replicas = func(i int32) *int32 { return &i }(4)
+				deployment.Spec.Template.Spec.NodeName = workerNodes[0].Name
+			},
+			expectedEvictedPodCount: 2,
+		},
+		{
+			description: "Evict Pod even Pods with local storage",
+			replicasNum: 5,
+			beforeFunc: func(deployment *appsv1.Deployment) {
+				deployment.Spec.Replicas = func(i int32) *int32 { return &i }(5)
+				deployment.Spec.Template.Spec.Volumes = []v1.Volume{
+					{
+						Name: "sample",
+						VolumeSource: v1.VolumeSource{
+							EmptyDir: &v1.EmptyDirVolumeSource{
+								SizeLimit: resource.NewQuantity(int64(10), resource.BinarySI)},
+						},
+					},
+				}
+			},
+			expectedEvictedPodCount: 2,
+		},
+	}
+	for _, tc := range tests {
+		t.Logf("Creating deployment %v in %v namespace", deploymentObj.Name, deploymentObj.Namespace)
+		tc.beforeFunc(deploymentObj)
+
+		_, err = clientSet.AppsV1().Deployments(deploymentObj.Namespace).Create(ctx, deploymentObj, metav1.CreateOptions{})
+		if err != nil {
+			t.Logf("Error creating deployment: %v", err)
+			if err = clientSet.AppsV1().Deployments(deploymentObj.Namespace).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+				LabelSelector: labels.SelectorFromSet(labels.Set(map[string]string{"app": "test-duplicate", "name": "test-duplicatePods"})).String(),
+			}); err != nil {
+				t.Fatalf("Unable to delete deployment: %v", err)
+			}
+			return
+		}
+		//defer clientSet.AppsV1().Deployments(deploymentObj.Namespace).Delete(ctx, deploymentObj.Name, metav1.DeleteOptions{})
```

Comment:
I would think this should be uncommented to prevent cascading failures if one test fails.

Reply:
Fix done. Would you please kindly review?